### PR TITLE
yara: add package

### PR DESCRIPTION
**Pull request by ratkaj**
Maintainer: Marko Ratkaj <marko.ratkaj@sartura.hr> / @ratkaj
Compile tested: LEDE master, mvebu
Run tested: LEDE master, mvebu, Linksys wrt3200

Description:
YARA is a tool aimed at (but not limited to) helping malware researchers
to identify and classify malware samples. With YARA you can create
descriptions of malware families based on textual or binary patterns.

### CODE REVIEW

**Comment by diizzyy** [comment]
You have some stray commits in your pull request, please resolve those.

File: utils/yara/Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=yara
+PKG_VERSION:=3.6.3
+PKG_RELEASE:=1
+PKG_LICENSE:=BSD-3-Clause
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline utils/yara/Makefile]: No, this will obviously clash eventually with something... ;-)

**Comment by diizzyy** [comment]
I would imagine that the magic module is of interest?

**Comment by ratkaj** [comment]
@diizzyy not only the magic module
Not sure what you mean, what can clash exactly?
v$(PKG_VERSION).tar.gz is the name of the file on the server (v3.6.3.tar.gz), is there a way to rename it on save?

Btw, i cleaned the stray commits.

**Comment by zorun** [comment]
Try this:

```
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/VirusTotal/yara/archive/v$(PKG_VERSION)/
```

**Comment by ratkaj** [comment]
It works, thank you

**Comment by diizzyy** [comment]
distfiles will eventually clash if we have a bunch of packages that uses files that are called v<version numer>.tar.gz not to mention it makes it impossible to distribute in a sane manner. I figured that you might want the functionality provided by --enable-magic out of the box but that might not be the case?

**Comment by ratkaj** [comment]
You are right, i should have enabled magic. I might add a config option so people can select the modules they need, with the depends they need.
--enable-dotnet  is probably useless on a gateway but has no depends, will leave it disabled
--enable-magic needs liibmagic
--enable-cuckoo needs jansson

**Comment by diizzyy** [comment]
When in doubt I personally use FreeBSD as a source, I find that they have reasonable tradeoffs in general.
https://svnweb.freebsd.org/ports/head/security/yara/Makefile?revision=445141&view=markup#l18
http://www.freshports.org/ is probably the easiest way to find a port.
I agree that dotnet doesn't make much sense in this case/distro.

**Pull request merged**